bc_obps/reporting/service/report_verification_service.py

```diff
@@ -53,3 +58,31 @@ def save_report_verification(version_id: int, data: ReportVerificationIn) -> Rep
         )
 
         return report_verification
+
+    @staticmethod
+    def get_report_needs_verification(version_id: int) -> bool:
+        """
+        Determines if a report needs verification data based on its purpose
+        and attributable emissions.
+        """
+        REGULATED_OPERATION_PURPOSES = {
+            Operation.Purposes.OBPS_REGULATED_OPERATION,
+            Operation.Purposes.OPTED_IN_OPERATION,
+            Operation.Purposes.NEW_ENTRANT_OPERATION,
+        }
+        ATTRIBUTABLE_EMISSION_THRESHOLD = Decimal("25000000")
+        registration_purpose = ReportAdditionalDataService.get_registration_purpose_by_version_id(version_id)
+
+        # Registration Purpose: verification data is required if the registration purpose is in REGULATED_OPERATION_PURPOSES
+        if isinstance(registration_purpose, dict):
+            registration_purpose = registration_purpose.get("registration_purpose", {})
```

Comment:
We're guaranteed this to be true by the additional data service.
We should also avoid reassigning a value of a different type to a variable
```suggestion
        registration_purpose_value = registration_purpose.get("registration_purpose", {})
```